lib/record_store/provider/ns1/client.rb

```diff
@@ -0,0 +1,42 @@
+require 'ns1'
+
+module RecordStore
+  class Provider::NS1 < Provider
+    class Error < StandardError; end
+    class Client < ::NS1::Client
+      def initialize(api_key:)
+        super(api_key)
+      end
+
+      def zones
+        super
+      end
+
+      def zone(name)
+        super(name)
+      end
+
+      def record(zone:, fqdn:, type:)
+        result = super(zone, fqdn, type)
+        return nil if result.is_a? NS1::Response::Error
+        result
+      end
+
+      def create_record(zone:, fqdn:, type:, params:)
+        result = super(zone, fqdn, type, params)
+        raise(Error, result.to_s) if result.is_a? NS1::Response::Error
+      end
```

Comment:
Maybe return `nil` explicitly at the end so it's clear this won't return anything ?

Reply:
Sure, it will be much clear that it won't return anything with explicit return `nil`.

```ruby
      def create_record(zone:, fqdn:, type:, params:)
        result = super(zone, fqdn, type, params)
        raise(Error, result.to_s) if result.is_a? NS1::Response::Error
        return nil
      end
```